src/lando/main/scm/git.py

```diff
@@ -229,7 +247,13 @@ def format_stack_amend(self) -> Optional[list[str]]:
 
     def format_stack_tip(self, commit_message: str) -> Optional[list[str]]:
         """Add an autoformat commit to the top of the patch stack."""
-        self._git_run("commit", "--all", "--message", commit_message, cwd=self.path)
+        try:
+            self._git_run("commit", "--all", "--message", commit_message, cwd=self.path)
+        except SCMException as exc:
+            if "nothing to commit, working tree clean" in exc.out:
+                return []
```

Comment:
Is there a more rigorous way of doing this check? Is there a specific error code we could be checking when raising `SCMException`? Looking at your implementation of `_git_run`, looks like we are raising this exception specifically based on a return code, and perhaps we shouldn't be raising this as an exception at all from the get go.

Reply:
The initial implementation had a conditional to allow throwing an exception or not, but I realised at the time that we always were throwing an exception, so I removed the parameter.

We could re-add it, but I can't find documentation about `git commit`'s error codes (at the moment it return `1` for this case), so we'd still have to inspect the error message to determine the cause of the error (FWIW, this is also how we do it for Mercurial) https://github.com/mozilla-conduit/lando/blob/main/src/lando/main/scm/hg.py#L372-L377